src/core/server/opensearch/opensearch_config.ts

```diff
@@ -334,6 +345,12 @@ export class OpenSearchConfig {
     this.password = rawConfig.password;
     this.customHeaders = rawConfig.customHeaders;
     this.disablePrototypePoisoningProtection = rawConfig.disablePrototypePoisoningProtection;
+    this.compression =
+      typeof rawConfig.compression === 'boolean' && rawConfig.compression
+        ? DEFAULT_COMPRESSION
+        : typeof rawConfig.compression === 'string'
+        ? rawConfig.compression
+        : undefined;
```

Comment:
An expression of `rawConfig.compression = false` gets out of this as `undefined` which is the wrong message out of this function (because it was defined). For the sanity of future devs, we should:
```suggestion
        : false;
```

Reply:
Also, for the sanity of future devs, can we not use nested ternaries, especially without parens? calling a function with readable if statements isn't the worst thing imo :)